Source/OxyPlot/Series/XYAxisSeries.cs

```diff
@@ -750,62 +750,65 @@ protected int UpdateWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter,
         /// <param name="targetX">target x.</param>
         /// <param name="initialGuess">initial guess index.</param>
         /// <returns>
-        /// index of x with max(x) &lt;= target x or -1 if cannot find
+        /// index of x with max(x) &lt;= target x or 0 if cannot find
         /// </returns>
-        protected int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
+        public int FindWindowStartIndex<T>(IList<T> items, Func<T, double> xgetter, double targetX, int initialGuess)
```

Comment:
Making it public to be able to call it from tests. Is there a better way?

Reply:
> Yes, please add an example that shows what is broken without your fix.
> There are some issues with code style in your PR. We are not super-religious about this, but some things we keep consistent (braces around single-line ifs and loops and spacing around operators for example). If you could take another look at this it would be great :).
> 
> Also, could `FindWindowStartIndex(...)` be rewritten to use `List.BinarySearch(...)`?

Sorry about the code style issues! I agree style should be consistent - my apologies!

I don't think List.BinarySearch would be a good fit, since we need to somehow handle the NaN-values. They can't be sorted on their own, we need to sort by their neighbors. But the IComparer in List.BinarySearch doesn't have access to the index, so it can't find the neighbors.

Reply:
This is the sort of thing I'd be tempted to move out of the class into Utilities somewhere as a `public static` method. There is no reason this logic could be used by e.g. annotations, one-dimensional series, pre-labelled axes, etc.. I'd be tempted to move everything to do with monotonicity optimisations into a new class.

Reply:
Yeah, I agree! I just wanted to keep my PR as minimal as possible, since it's meant as a bug fix. 

But I also noted that the function could/should have been a static method in some other class :-)